aws/signer/v4/v4_test.go

```diff
@@ -201,9 +201,8 @@ func TestPresignEmptyBodyS3(t *testing.T) {
 	req, body := buildRequest("s3", "us-east-1", "hello")
 	signer := buildSigner()
 	signer.Presign(req, body, "s3", "us-east-1", 5*time.Minute, time.Now())
-	hash := req.Header.Get("X-Amz-Content-Sha256")
-	if e, a := "UNSIGNED-PAYLOAD", hash; e != a {
-		t.Errorf("expect %v, got %v", e, a)
+	if a := req.Header.Get("X-Amz-Content-Sha256"); len(a) != 0 {
```

Comment:
Do we have a test where the service isn't `"s3"`? To ensure it is included during presigning

Reply:
Added additional tests to cover these cases.